packages/react-bootstrap-table2/test/row.test.js

```diff
@@ -442,128 +442,180 @@ describe('Row', () => {
+        });
+
+        describe('when selectRow.hideSelectColumn is true', () => {
+          beforeEach(() => {
+            selectRow = { mode: 'checkbox', hideSelectColumn: true };
+            wrapper = shallow(
+              <Row
+                { ...mockBodyResolvedProps }
+                rowIndex={ rowIndex }
+                columns={ defaultColumns }
+                row={ row }
+                selectRow={ selectRow }
+              />);
+          });
+
+          it('should not render <SelectionCell />', () => {
+            expect(wrapper.find(SelectionCell).length).toBe(0);
+          });
+        });
       });
 
-      it('should render Row component successfully with onClick event', () => {
-        expect(wrapper.length).toBe(1);
-        expect(wrapper.find('tr').prop('onClick')).toBeDefined();
+      describe('if selectable prop is false', () => {
+        beforeEach(() => {
+          selectRow = { mode: 'checkbox' };
+          wrapper = shallow(
+            <Row
+              { ...mockBodyResolvedProps }
+              rowIndex={ rowIndex }
+              columns={ defaultColumns }
+              row={ row }
+              keyField={ keyField }
+              selectRow={ selectRow }
+              selectable={ false }
+            />);
+        });
+
+        it('should render SelectionCell component with correct disable prop correctly', () => {
+          expect(wrapper.find(SelectionCell).length).toBe(1);
+          expect(wrapper.find(SelectionCell).prop('disabled')).toBeTruthy();
+        });
       });
-    });
 
-    describe('if selectRow.hideSelectColumn is true', () => {
-      beforeEach(() => {
-        selectRow.hideSelectColumn = true;
-        wrapper = shallow(
-          <Row
-            { ...mockBodyResolvedProps }
-            rowIndex={ rowIndex }
-            columns={ defaultColumns }
-            row={ row }
-            selectRow={ selectRow }
-            selected
-            selectable
-          />);
+      describe('if selectable prop is true', () => {
+        selectRow = { mode: 'checkbox' };
+
+        beforeEach(() => {
+          wrapper = shallow(
+            <Row
+              { ...mockBodyResolvedProps }
+              rowIndex={ rowIndex }
+              columns={ defaultColumns }
+              row={ row }
+              keyField={ keyField }
+              selectRow={ selectRow }
+              selectable
+            />);
+        });
+
+        it('should render SelectionCell component with correct disable prop correctly', () => {
+          expect(wrapper.find(SelectionCell).length).toBe(1);
+          expect(wrapper.find(SelectionCell).prop('disabled')).toBeFalsy();
+        });
       });
 
-      it('should render Row component without selection column', () => {
-        expect(wrapper.length).toBe(1);
-        expect(wrapper.find(SelectionCell).length).toBe(0);
+      describe('if selectRow.clickToSelect is true', () => {
+        selectRow = { mode: 'checkbox' };
```

Comment:
could you please make this declare more consistency? You put it in `beforeEach` or the `describe`....  Not consistency

Reply:
Yes it does. I planned to prevent declaring `selectRow` again and again but missing consistency issue. Thanks for your advice 💯